wordpress-mobile/WordPress-FluxC-Android
# Update Product Addon remote and local models

Follow-up to https://github.com/woocommerce/woocommerce-android/issues/4405

### Description

Working longer with Add-on API and domain made me understand it more and I think we can improve our models for:
- storing less fields in database, which reduces its complexity
- improve nullability of remote object fields to not worry about cases that, according to API, don't happen

### To test

This is not a testable PR but please take a second look at api description to double-check me if I didn't make any mistake. I'll try to provide also a rich description in this PR of decision process behind changed.
